### PR TITLE
Add logout buttons to samples

### PR DESCRIPTION
**Pull request by fredrik-lundin**
This PR implements the logout buttons mentioned in #11

It includes changes that:
1. Adds logout button and functionality to the `MVC Client Sample`
2. Adds logout button and functionality to the `Identity Server Sample` 
3. Fixes a bug that prevented the user to perform more than one login attempt in a row in dev mode
4. Adds test to verify the bug fix mentioned above

### CODE REVIEW

File: samples/IdentityServerSample/Controllers/AccountController.cs [hunk 1]
```diff
@@ -71,5 +71,11 @@ public async Task<IActionResult> ExternalLoginCallback()
 
             return Redirect("~/");
         }
+
+        public async Task<IActionResult> Logout()
```
Comment (PeterOrneholm) [inline samples/IdentityServerSample/Controllers/AccountController.cs]: Should use HttpPost.

File: samples/IdentityServerSample/Views/Shared/_Layout.cshtml [hunk 2]
```diff
@@ -23,6 +23,10 @@
     <nav class="navbar navbar-dark bg-dark fixed-top">
         <div class="container">
             <a asp-area="" asp-controller="Home" asp-action="Index" class="navbar-brand">ActiveLogin - IdentityServerSample</a>
+            @if (User.Identity?.IsAuthenticated == true)
+            {
+                <a asp-area="" asp-controller="Account" asp-action="Logout" class="btn btn-primary">Logout</a>
```
Comment (PeterOrneholm) [inline samples/IdentityServerSample/Views/Shared/_Layout.cshtml]: Should be a form with HttpPost action.

File: samples/MvcClientSample/Controllers/HomeController.cs [hunk 3]
```diff
@@ -12,5 +12,10 @@ public IActionResult Index()
                 Claims = User.Claims
             });
         }
+
+        public IActionResult Logout()
```
Comment (PeterOrneholm) [inline samples/MvcClientSample/Controllers/HomeController.cs]: Should be HttpPost.

File: samples/MvcClientSample/Views/Shared/_Layout.cshtml [hunk 4]
```diff
@@ -16,6 +16,10 @@
     <nav class="navbar  navbar-dark bg-primary fixed-top">
         <div class="container">
             <a asp-area="" asp-controller="Home" asp-action="Index" class="navbar-brand">ActiveLogin - MvcClientSample</a>
+            @if (User.Identity?.IsAuthenticated == true)
+            {
+                <a asp-area="" asp-controller="Home" asp-action="Logout" class="btn btn-primary">Logout</a>
```
Comment (PeterOrneholm) [inline samples/MvcClientSample/Views/Shared/_Layout.cshtml]: Should be form with HttpPost action.

File: samples/MvcClientSample/Controllers/HomeController.cs (hunk 3 above)
Comment (fredrik-lundin) [reply samples/MvcClientSample/Controllers/HomeController.cs]: Done

**Comment by fredrik-lundin** [comment]
@PeterOrneholm  Fixed according to feedback. I agree with using http post for logout.

Also fixed proper post logout redirect from IdSrv back to client

**Pull request merged**